src/Compatibility/Core/src/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -480,7 +470,7 @@ void ValidateNavbarExists(Page newCurrentPage)
 			//if the last time we did ViewDidLayoutSubviews we had other value for _hasNavigationBar
 			//we will need to relayout. This is because Current is updated async of the layout happening
 			if (_hasNavigationBar != NavigationPage.GetHasNavigationBar(newCurrentPage))
-				ViewDidLayoutSubviews();
+				View.InvalidateMeasure(Element);
```

Comment:
call `ViewDidLayoutSubviews` yourself doesn't seem like the best idea.  I've tested hiding and showing the navbar a few different ways and invalidating the measure seems to do the job